src/poll.h

```diff
@@ -0,0 +1,66 @@
+/**
+ * @file poll.h
+ * @brief API to poll files.
+ * @author Simon Nagl
+ */
+
+#ifndef NETDATA_POLL_H
+#define NETDATA_POLL_H
+
+#include <time.h>
+
+#define POLLIN 1
+#define POLLOUT 2
+#define POLLERR 3
+
+/**
+ * @brief Thread loop.
+ *
+ * This must only be called once by the main thread.
+ *
+ * @param ptr TODO
+ *
+ * @return TODO
```

Comment:
For the description how to use return value for pthreads.

Or should we just remove the detailed description from this methods because both values are not used.